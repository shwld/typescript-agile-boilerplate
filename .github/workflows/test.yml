name: test

on:
  push:
    branches: ["main"]
  pull_request:
    types: [opened, synchronize]

jobs:
  build:
    name: Test
    timeout-minutes: 15
    runs-on: ${{ matrix.os }}
    # To use Remote Caching, uncomment the next lines and follow the steps below.
    # env:
    #  TURBO_TOKEN: ${{ secrets.TURBO_TOKEN }}
    #  TURBO_TEAM: ${{ secrets.TURBO_TEAM }}
    strategy:
      matrix:
        # os: [ubuntu-latest, macos-latest]
        os: [ubuntu-latest]
    services:
      # Label used to access the service container
      postgres:
        # Docker Hub image
        image: postgres
        # Provide the password for postgres
        env:
          POSTGRES_USER: postgres
          POSTGRES_PASSWORD: password
          POSTGRES_DB: app_test
        # Set health checks to wait until postgres has started
        options: >-
          --health-cmd pg_isready
          --health-interval 10s
          --health-timeout 5s
          --health-retries 5

    steps:
      - name: Check out code
        uses: actions/checkout@v2
        with:
          fetch-depth: 2

      - name: Setup Node.js environment
        uses: actions/setup-node@v2
        with:
          node-version: 16
          cache: 'yarn'

      - name: Setup Test
        run: |
          echo 'DATABASE_URL="postgresql://postgres:password@postgres:5432/app_test?schema=public"' > ./infrastructures/db/.env
          echo 'DATABASE_URL="postgresql://postgres:password@postgres:5432/app_test?schema=public"' > ./use-cases/graphql-resolvers/.env
          yarn db:migrate

      - name: Install dependencies
        run: yarn

      - name: Test
        run: yarn test
