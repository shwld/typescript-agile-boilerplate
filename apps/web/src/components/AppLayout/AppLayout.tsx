import {
  Box,
  Flex,
  Avatar,
  HStack,
  IconButton,
  Menu,
  MenuButton,
  MenuList,
  MenuItem,
  useDisclosure,
  useColorModeValue,
  Stack,
  Button,
} from '@chakra-ui/react';
import Link from 'next/link';
import { HamburgerIcon, CloseIcon } from '@chakra-ui/icons';
import { signOut } from 'next-auth/react';

type ProjectLayoutProps = {
  children: React.ReactNode;
};

const ProjectLayout = ({ children }: ProjectLayoutProps) => {
  const { isOpen, onOpen, onClose } = useDisclosure();
  return (
    <>
      <Box bg={useColorModeValue('blue.200', 'gray.900')} px={4}>
        <Flex h={12} alignItems={'center'} justifyContent={'space-between'}>
          <IconButton
            size={'md'}
            icon={isOpen ? <CloseIcon /> : <HamburgerIcon />}
            aria-label={'Open Menu'}
            display={{ md: 'none' }}
            onClick={isOpen ? onClose : onOpen}
          />
          <HStack spacing={8} alignItems={'center'}>
            <Box>Logo</Box>
            <HStack
              as={'nav'}
              spacing={4}
              display={{ base: 'none', md: 'flex' }}
            >
              <Link href="/accounts">ACCOUNTS</Link>
              <Link href="/projects">PROJECTS</Link>
            </HStack>
          </HStack>
          <Flex alignItems={'center'}>
            <Menu>
              <MenuButton
                as={Button}
                rounded={'full'}
                variant={'link'}
                cursor={'pointer'}
                minW={0}
              >
                <Avatar
                  size={'sm'}
                  src={
                    'https://images.unsplash.com/photo-1493666438817-866a91353ca9?ixlib=rb-0.3.5&q=80&fm=jpg&crop=faces&fit=crop&h=200&w=200&s=b616b2c5b373a80ffc9636ba24f7a4a9'
                  }
                />
              </MenuButton>
              <MenuList>
                <MenuItem>
                  <Link href="/accounts">ACCOUNTS</Link>
                </MenuItem>
                <MenuItem>
                  <Link href="/projects">PROJECTS</Link>
                </MenuItem>
                <MenuItem>
                  <Button onClick={() => signOut()}>Logout</Button>
                </MenuItem>
              </MenuList>
            </Menu>
          </Flex>
        </Flex>

        {isOpen ? (
          <Box pb={4} display={{ md: 'none' }}>
            <Stack as={'nav'} spacing={4}></Stack>
          </Box>
        ) : null}
      </Box>

      {children}
    </>
  );
};

export default ProjectLayout;
