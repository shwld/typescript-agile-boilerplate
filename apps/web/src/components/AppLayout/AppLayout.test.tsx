import AppLayout from './AppLayout';
import { render } from '@testing-library/react';
import { MockedUrqlProvider } from '~/test/MockedUrqlProvider';

describe('AppLayout', () => {
  const renderComponent = () => {
    const renderResult = render(
      <MockedUrqlProvider>
        <AppLayout>test</AppLayout>
      </MockedUrqlProvider>
    );
    return renderResult;
  };
  test('Snapshot', () => {
    expect(renderComponent().asFragment()).toMatchSnapshot();
  });
  test('success', () => {
    const { getByText } = renderComponent();
    expect(getByText('test')).toBeTruthy();
  });
});
