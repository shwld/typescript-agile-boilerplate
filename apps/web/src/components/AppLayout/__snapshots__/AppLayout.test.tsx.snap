// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout Snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-suwfoa"
  >
    <div
      class="css-67ntpa"
    >
      <button
        aria-label="Open Menu"
        class="chakra-button css-1mh58fi"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
            fill="currentColor"
          />
        </svg>
      </button>
      <div
        class="chakra-stack css-16awuyr"
      >
        <div
          class="css-0"
        >
          Logo
        </div>
      </div>
      <div
        class="css-70qvj9"
      >
        <button
          aria-controls="menu-list-5"
          aria-expanded="false"
          aria-haspopup="menu"
          class="chakra-button chakra-menu__menu-button css-1t30i5w"
          id="menu-button-5"
          type="button"
        >
          <span
            class="css-xl71ch"
          >
            <span
              class="chakra-avatar css-1ruhx4w"
            >
              <svg
                aria-label=" avatar"
                class="chakra-avatar__svg css-16ite8i"
                role="img"
                viewBox="0 0 128 128"
              >
                <path
                  d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                  fill="currentColor"
                />
                <path
                  d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                  fill="currentColor"
                />
              </svg>
            </span>
          </span>
        </button>
        <div
          class="css-0"
          style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
        >
          <div
            aria-orientation="vertical"
            class="chakra-menu__menu-list css-sjt5nk"
            id="menu-list-5"
            role="menu"
            style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
            tabindex="-1"
          >
            <button
              class="chakra-menu__menuitem css-1k756t9"
              data-index="0"
              id="menu-list-5-menuitem-1"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <a
                href="/accounts"
              >
                ACCOUNTS
              </a>
            </button>
            <button
              class="chakra-menu__menuitem css-1k756t9"
              data-index="1"
              id="menu-list-5-menuitem-2"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <a
                href="/projects"
              >
                PROJECTS
              </a>
            </button>
            <button
              class="chakra-menu__menuitem css-1k756t9"
              data-index="2"
              id="menu-list-5-menuitem-3"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <a>
                Logout
              </a>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  test
</DocumentFragment>
`;
